hmac-mapper/src/main/java/de/intension/protocol/oidc/mappers/HmacPairwiseEmailMapper.java

```diff
@@ -0,0 +1,115 @@
+package de.intension.protocol.oidc.mappers;
+
+import java.util.List;
+
+import org.keycloak.models.ClientSessionContext;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.ProtocolMapperModel;
+import org.keycloak.models.UserSessionModel;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.representations.AccessToken;
+import org.keycloak.representations.IDToken;
+import org.keycloak.utils.StringUtil;
+
+/**
+ * Pairwise identifier mapper using
+ * <a href="https://datatracker.ietf.org/doc/html/rfc2104">HMAC</a>.
+ * This OIDC mapper will replace the {@code email} field in the token with a
+ * HMAC-hashed email
+ * instead of the email. Sector identifier is mandatory for this mapper and must
+ * have a valid hostname.
+ *
+ * @see <a href=
+ *      "https://docs.oracle.com/en/java/javase/11/docs/specs/security/standard-names.html#mac-algorithms">mac-algorithms</a>
+ */
+public class HmacPairwiseEmailMapper extends HmacPairwiseSubMapper {
+
+    public static final String PROVIDER_ID = "oidc-hmac-pairwise-email-mapper";
+
+    private static final String EMAIL_DOMAIN_PROP_NAME = "emailDomain";
+    private static final String EMAIL_DOMAIN_PROP_LABEL = "Email domain";
+    private static final String EMAIL_DOMAIN_PROP_HELP = "The email domain is appended to the claim value after pseudonymization.";
+
+    @Override
+    public String getId() {
+        return PROVIDER_ID;
+    }
+
+    @Override
+    public String getDisplayType() {
+        return "HMAC pairwise email";
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Updates the email claim in tokens with a pseudonymized value from a selected attribute.";
+    }
+
+    @Override
+    public IDToken transformIDToken(IDToken token, ProtocolMapperModel mappingModel, KeycloakSession session,
```

Comment:
I thought this should only be done if it is not set. Ticket says it should, and not must but ...

Reply:
Added configuration for this - forgot about the OIDCAttributeMapperHelper